### PR TITLE
feat: update available social targets after connector deletion

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Update available social targets after connector deletion.

## Testing

Passed UTs, ITs and tested locally.

### CODE REVIEW

File: packages/core/src/routes/connector.ts [hunk 1]
```diff
@@ -277,6 +277,14 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
 
       await deleteConnectorById(id);
 
+      try {
+        const logtoConnector = await getLogtoConnectorById(id);
+
+        if (logtoConnector.type === ConnectorType.Social) {
+          await removeUnavailableSocialConnectorTargets();
+        }
+      } catch {}
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:286]: wondering the reason to swallow exceptions?

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -277,6 +277,14 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
 
       await deleteConnectorById(id);
 
+      try {
+        const logtoConnector = await getLogtoConnectorById(id);
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:281]: The connector has been deleted, this will fail?

Comment (darcyYe) [reply packages/core/src/routes/connector.ts]: Fixed the logic.

File: packages/core/src/routes/connector.ts [hunk 3]
```diff
@@ -275,8 +276,20 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         params: { id },
       } = ctx.guard;
 
+      const { connectorId } = await findConnectorById(id);
+      const connectorFactories = await loadConnectorFactories();
+      const connectorFactory = connectorFactories.find(
+        ({ metadata }) => metadata.id === connectorId
+      );
+
+      console.log(connectorId, connectorFactories);
```
Comment (simeng-li) [inline packages/core/src/routes/connector.ts:285]: console

**Review by simeng-li — CHANGES_REQUESTED** [review]

**Review by wangsijie — APPROVED** [review]
Overall LGTM, remember to remove the console.log

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**